clockdiff.c

```diff
@@ -198,20 +198,19 @@ static int measure_inner_loop(struct run_state *ctl, struct measure_vars *mv)
 	long histime1 = 0;
 	long recvtime;
 	long sendtime;
+	struct pollfd p = { .fd = ctl->sock_raw, .events = POLLIN | POLLHUP };
 
-	FD_ZERO(&mv->ready);
-	FD_SET(ctl->sock_raw, &mv->ready);
 	{
 		long tmo = ctl->rtt + ctl->rtt_sigma;
 
 		mv->tout.tv_sec = tmo / 1000;
-		mv->tout.tv_usec = (tmo - (tmo / 1000) * 1000) * 1000;
+		mv->tout.tv_nsec = (tmo - (tmo / 1000) * 1000) * 1000000;
 	}
 
-	if ((mv->count = select(FD_SETSIZE, &mv->ready, NULL, NULL, &mv->tout)) <= 0)
+	if ((mv->count = ppoll(&p, 1, &mv->tout, NULL)) <= 0)
 		return BREAK;
 
-	gettimeofday(&mv->tv1, NULL);
+	clock_gettime(CLOCK_REALTIME, &mv->ts1);
```

Comment:
`CLOCK_REALTIME` requires `CAP_SYS_TIME`. But I guess all places which use `iputil_srand_fallback()` also require privileges (at least ninfod and `niquery_init_nonce()` in `ping`, I suppose `rdisc` needs privileges as well).

Reply:
CAP_SYS_TIME is required when setting the time. Reading is unprivileged operation.

http://man7.org/linux/man-pages/man7/capabilities.7.html see CAP_SYS_TIME

Reply:
ok. Acked from my side anyway.